sdk/eventhub/Microsoft.Azure.EventHubs/src/Primitives/AsyncLock.cs

```diff
@@ -14,7 +14,7 @@ namespace Microsoft.Azure.EventHubs
     /// </summary>
     public class AsyncLock : IDisposable
     {
-        readonly SemaphoreSlim asyncSemaphore = new SemaphoreSlim(1);
+        readonly SemaphoreSlim asyncSemaphore = new SemaphoreSlim(1, 1);
```

Comment:
Any chance that you'd consider marking this `internal` as well?   I can't think of a scenario where it should be exposed to callers as part of the public API surface of the package.